bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableDataSettingsFactory.java

```diff
@@ -0,0 +1,265 @@
+    builder.sampleRowKeysSettings()
+        .setRetrySettings(defaultRetrySettings(options));
+  }
+
+  /**
+   * To build mutateRowSettings with default Settings based on retries setting.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static void buildMutateRow(Builder builder, BigtableOptions options) {
+    builder.mutateRowSettings()
+        .setRetrySettings(defaultRetrySettings(options));
+  }
+
+  /**
+   * To build readRows with default Settings based on retries setting.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static void buildReadRows(Builder builder, BigtableOptions options) {
+    // TODO: set readPartialRowTimeout for watchdog timer, taken BigtableSession#setupWatchdog()
+    builder.readRowsSettings()
+        .setRetrySettings(defaultRetrySettings(options));
+  }
+
+  /**
+   * This method builds ReadModifyWrite with no retry configurations
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static void buildReadModifyWrite(Builder builder, long rpcTimeoutMs) {
+    builder.readModifyWriteRowSettings()
+        .setSimpleTimeoutNoRetries(ofMillis(rpcTimeoutMs));
+  }
+
+  /**
+   * This method builds CheckAndMutateRow with no retry configurations
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static void buildCheckAndMutateRow(Builder builder, long rpcTimeoutMs) {
+    builder.checkAndMutateRowSettings()
+        .setSimpleTimeoutNoRetries(ofMillis(rpcTimeoutMs));
+  }
+
+  /**
+   * To create default RetrySettings, for BigtableDataSettings.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static RetrySettings defaultRetrySettings(BigtableOptions options) {
+    RetryOptions retryOptions = options.getRetryOptions();
+
+    RetrySettings.Builder retryBuilder = RetrySettings.newBuilder()
+        .setInitialRetryDelay(ofMillis(retryOptions.getInitialBackoffMillis()))
+        .setRetryDelayMultiplier(retryOptions.getBackoffMultiplier())
+        .setMaxRetryDelay(ofMillis(retryOptions.getMaxElapsedBackoffMillis()))
+        .setMaxAttempts(retryOptions.getMaxScanTimeoutRetries());
+
+    // configurations for RPC timeouts
+    retryBuilder
+        .setInitialRpcTimeout(ofMillis(options.getCallOptionsConfig().getShortRpcTimeoutMs()))
+        .setMaxRpcTimeout(ofMillis(retryOptions.getReadPartialRowTimeoutMillis()))
+        .setTotalTimeout(ofMillis(options.getCallOptionsConfig().getLongRpcTimeoutMs()));
+
+    // TODO: an option to set RetryOptions#allowRetriesWithoutTimestamp
```

Comment:
I dont think this will ever be supported as a setting. Please throw an unsupported exception instructing users to use ajay's unsafe Mutation factory method